packages/spl-token/src/parsers/event.ts

```diff
@@ -0,0 +1,300 @@
+          toBalance,
+          toAccount,
+          toOwner,
+          mint,
+        }
+      }
+      case SPLTokenEventType.SetAuthority: {
+        const { account, authority, authorityType, newAuthority } = parsed.info
+        const balance = getTokenBalance(parentTx, account) as string
+        const { mint, owner } = await this.getMintAndOwner(account)
+
+        return {
+          id,
+          timestamp,
+          type,
+          balance,
+          account,
+          owner,
+          newOwner: newAuthority,
+          authorityType,
+          mint,
+        }
+      }
+      case SPLTokenEventType.Approve: {
+        const { source: account, owner, delegate, amount } = parsed.info
+        const balance = getTokenBalance(parentTx, account) as string
+        const { mint } = await this.getMintAndOwner(account)
+
+        return {
+          id,
+          timestamp,
+          type,
+          amount,
+          balance,
+          account,
+          owner,
+          delegate,
+          mint,
+        }
+      }
+      case SPLTokenEventType.ApproveChecked: {
+        const { source: account, owner, delegate, tokenAmount } = parsed.info
+        const balance = getTokenBalance(parentTx, account) as string
+        const { mint } = await this.getMintAndOwner(account)
+
+        return {
+          id,
+          timestamp,
+          type: SPLTokenEventType.Approve,
+          amount: tokenAmount.amount,
+          balance,
+          account,
+          owner,
+          delegate,
+          mint,
+        }
+      }
+      case SPLTokenEventType.Revoke: {
+        const { source: account, owner } = parsed.info
+        const balance = getTokenBalance(parentTx, account) as string
+        const { mint } = await this.getMintAndOwner(account)
+
+        return {
+          id,
+          timestamp,
+          type,
+          balance,
+          account,
+          owner,
+          mint,
+        }
+      }
+      case SPLTokenEventType.SyncNative: {
+        const { account } = parsed.info
+        const balance = getTokenBalance(parentTx, account) as string
+        const { mint, owner } = await this.getMintAndOwner(account)
+
+        return {
+          id,
+          timestamp,
+          type,
+          balance,
+          account,
+          owner,
+          mint,
+        }
+      }
+
+      default: {
+        console.log('NOT PARSED IX TYPE', (parsed as any).type)
+        console.log(id)
+        return
+      }
+    }
+  }
+
+  protected async getMintAndOwner(account: string): Promise<MintOwner> {
+    const dbEvent = await this.fetchMintDAL.getFirstValueFromTo(
+      [account],
+      [account],
+    )
+    if (dbEvent) {
+      const event = dbEvent.payload.event
+      const data = getMintAndOwnerFromEvent(event, account)
+
+      if (data) {
+        return {
+          mint: data.mint,
+          owner: data.owner || '',
+        }
+      }
+    }
+
+    return {
+      mint: '',
```

Comment:
We need to ensure that we will always have the mint here, by setting it mandatory on the discovery

Reply:
We ensure it now, but on this method, because we search it on the database and there's the case that we don't find it.